src/encode.c

```diff
@@ -904,7 +904,7 @@ PyImaging_LibTiffEncoderNew(PyObject *self, PyObject *args) {
                     &encoder->state, (ttag_t)key_int, (UINT16)PyLong_AsLong(value));
             } else if (type == TIFF_LONG) {
                 status = ImagingLibTiffSetField(
-                    &encoder->state, (ttag_t)key_int, (UINT32)PyLong_AsLong(value));
+                    &encoder->state, (ttag_t)key_int, (uint64_t)PyLong_AsLong(value));
```

Comment:
I don't fully understand this change and can't look into it right now (I'm not sure why the `TIFF_LONG8` error was triggered when this change is for `TIFF_LONG`), but at least on Windows, the `PyLong_AsLong` function returns the equivalent of `int32_t`. I think `PyLong_AsLongLong` should return `int64_t` on all platforms, although I'm not sure if it matters here.

To be clear, using `PyLong_AsLong` here instead of `PyLong_AsLongLong` is not a security issue, I'm just not sure whether it could save the wrong values.

Reply:
Sure, that seems better than casting the value. I've updated the commit.

Reply:
It returns `int64_t`, not `uint64_t`. It probably doesn't matter due to how signedness casting works, but it's not the same type.